### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by cancerberoSgx**
https://github.com/dsherret/ts-morph/issues/728

### CODE REVIEW

File: src/compiler/ast/common/Node.ts [hunk 1]
```diff
@@ -1299,15 +1299,16 @@ export class Node<NodeType extends ts.Node = ts.Node> {
     }
 
     /**
-     * Gets the line number at the start of the node.
+     * Gets the line number (1-indexed) of the start of the node in the *full* text of its source file.
```
Comment (dsherret) [inline src/compiler/ast/common/Node.ts]: Thanks! This is an improvement, but let's remove the emphasis (as nothing makes sense other than full text) and use "at" for both of these.

```suggestion
     * Gets the line number (1-indexed) at the start of the node in the full text of its source file. 
```

File: src/compiler/ast/common/Node.ts [hunk 2]
```diff
@@ -1299,15 +1299,16 @@ export class Node<NodeType extends ts.Node = ts.Node> {
...
+     * Gets the line number (1-indexed) of the start of the node in the *full* text of its source file. 
      * @param includeJsDocComments - Whether to include the JS doc comments or not.
      */
     getStartLineNumber(includeJsDocComments?: boolean) {
         return StringUtils.getLineNumberAtPos(this._sourceFile.getFullText(), this.getStartLinePos(includeJsDocComments));
     }
 
     /**
-     * Gets the line number of the end of the node.
+     * Gets the line number (1-indexed) of the end of the node in the *full* text of its source file.
```
Comment (dsherret) [inline src/compiler/ast/common/Node.ts]:
```suggestion
     * Gets the line number (1-indexed) at the end of the node in the full text of its source file. 
```

File: src/utils/TypeGuards.ts [hunk 3]
```diff
@@ -857,7 +858,9 @@ export class TypeGuards {
     /**
      * Gets if the node is a InferKeyword.
      */
-    static readonly isInferKeyword: (node: compiler.Node) => node is compiler.Node = TypeGuards.is(SyntaxKind.InferKeyword);
+    static isInferKeyword(node: compiler.Node): node is compiler.Node<Token<SyntaxKind.InferKeyword>> { 
+        return node.getKind() === SyntaxKind.InferKeyword;
+    }
```
Comment (dsherret) [inline src/utils/TypeGuards.ts]: This file is actually code generated so we can't make this change here. For example, this PR did the change to make everything in here use `TypeGuards.is`: https://github.com/dsherret/ts-morph/pull/703/files

Do you want to merge the jsdoc change above separately as that's a good change? Then we could come up for a fix for this separately. Let me know if you'd like to look at fixing the generated code or I can take a look at it soon.

Comment (cancerberoSgx) [reply src/utils/TypeGuards.ts]: I saw the note in the header comment but thought only applied to the actual method is()

I will fix there then. 

do you agree with that type casting ?

Comment (dsherret) [reply src/utils/TypeGuards.ts]: Oh yeah, I can see how that header comment might be confusing. Feel free to fix it!

Do you mean casting the result of `TypeGuards.is(SyntaxKind.InferKeyword)` to `compiler.Node<Token<SyntaxKind.InferKeyword>>`? That should be fine!

Thanks again! 😊

**Comment by dsherret** [comment]
Hmmm... I think github automatically closed this (maybe because the folder structure has changed so drastically). Feel free to reopen later!

**Pull request closed without merging**